bolt/test/dynamic-relocs.test

```diff
@@ -0,0 +1,13 @@
+// This test checks that BOLT creates entry points for addresses
+// referenced by dynamic relocations.
+
+RUN: %clang %cflags %S/Inputs/indirect_goto.c -o %t.exe -Wl,-q
```

Comment:
Let's explicitly build and test PIE and non-PIE variants.

Reply:
Added PIE and non-PIE variants.